Analysis/Tasks/PWGHF/HFCandidateCreator3Prong.cxx

```diff
@@ -154,11 +154,23 @@ struct HFCandidateCreator3ProngMC {
   {
     int8_t sign = 0;
     int8_t result = N3ProngDecays;
+    int8_t DecayChannel = 0;
+    int8_t size;
+    std::vector<int> arrDaughIndex;
+    std::vector<int> arrPDGDaugh;
+    std::vector<int> arrPDGResonant1 = {2212, 313};
+    std::vector<int> arrPDGResonant2 = {2224, 321};
+    std::vector<int> arrPDGResonant3 = {3124, 211};
 
     // Match reconstructed candidates.
     for (auto& candidate : candidates) {
       //Printf("New rec. candidate");
       result = N3ProngDecays;
+      DecayChannel = 0;
+      size = 0;
+      arrDaughIndex = {};
+      arrPDGDaugh = {0, 0, 0};
```

Comment:
Why does this need 3 elements if it is used only for daughters of the resonant channels?

Reply:
Because it is filled also for the non resonant case.